File: lab3/q5b.py
import random
import sys

line_counter = 0
matrix = []
data_list = []
elements = 0
target_rows = []
target_col = []

for current_line in sys.stdin:

    line_counter = line_counter + 1

    if line_counter == 1:
        row_count = int(current_line.strip())

    elif line_counter == 2: 
        col_count = int(current_line.strip())
        
        elements = row_count * col_count

    elif line_counter <= elements + 2:
        value = float(current_line.strip())
        data_list.append(value)

    elif line_counter >  elements + 2 :
        if current_line != "/n":

            if (elements + 2) % 2 == 0:
                if line_counter % 2 != 0:
                    row_target =  int(current_line.strip())
                    if  row_target > row_count or row_target < 0:
                        target_rows.append(-1)
                    else:    
                        target_rows.append(row_target)
                else:
                    col_target =  int(current_line.strip())
                    if  col_target > col_count or col_target < 0:
                        target_col.append(-1)
                    else:   
                        target_col.append(col_target)

            elif (elements + 2) % 2 != 0:
                if line_counter % 2 == 0:
                    row_target =  int(current_line.strip())
                    if  row_target > row_count or row_target < 0:
                        target_rows.append(-1)
                    else:    
                        target_rows.append(row_target)
                else:
                    col_target =  int(current_line.strip())
                    if  col_target > col_count or col_target < 0:
                        target_col.append(-1)
                    else:   
                        target_col.append(col_target)
                

#Making a Matrix with list
if abs(row_count - col_count) == 1:
    for i in range(row_count):
        rowList = []
        for j in range (col_count):
            rowList.append(data_list[((row_count+1) * i) + j])
        matrix.append(rowList)

else:
    for i in range(row_count):
        rowList = []
        for j in range (col_count):
            rowList.append(data_list[((row_count) * i) + j])
        matrix.append(rowList)


print("Matrix:")
#Display Matrix by row and coloumn
for i in range (row_count):
    for j in range(col_count):
        print(matrix[i][j], end=" ")
    print("")


print("\nTargets:")
for i in range (len(target_rows)):
    row = target_rows[i]-1
    col = target_col[i]-1
    if (target_rows[i] == -1) or (target_col[i] == -1):
        print("Target", str(i+1), ": ERROR TARGET NOT IN MATRIX ")
    else:
        print("Target", str(i+1) + ": Matrix[" + str(row+1) + "][" + str(col+1) + "] = " + str(matrix[row][col]))
 

File: lab3/q5b.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;


auto main() -> int
{
    auto line_counter {0};
    auto elements {0};
    auto row_count{0};
    auto col_count{0};

    std::vector<vector<float>> matrix = {};
    std::vector<float> data_list = {};
    std::vector<int> target_rows = {};
    std::vector<int> target_col = {};

    std::string lineInput;

    while ( getline( std::cin, lineInput ) )
	{
         line_counter = line_counter + 1;

        if (line_counter == 1){
            row_count = std::stoi(lineInput);
         }
         else if (line_counter == 2){
            col_count = std::stoi(lineInput);
            elements = row_count * col_count;
         }
         else if (line_counter <= elements + 2){
            float value =  std::stof(lineInput);
            data_list.push_back(value);
         }
         else if(line_counter > elements + 2){
             if(!lineInput.empty()){

                 if((elements + 2) % 2 == 0){

                     if (line_counter % 2 != 0){
                        auto row_target = std::stoi(lineInput);
                        if ((row_target > row_count) || row_target < 0){
                            target_rows.push_back(-1);
                        } else {
                            target_rows.push_back(row_target);
                        }
                     } else {
                         auto col_target = std::stoi(lineInput);
                         if ((col_target > col_count) || col_target < 0){
                            target_col.push_back(-1);
                        } else {
                            target_col.push_back(col_target);
                        }
                     }
                 } else if ((elements + 2) % 2 != 0){

                     if (line_counter % 2 == 0){
                        auto row_target = std::stoi(lineInput);
                        if ((row_target > row_count) || row_target < 0){
                            target_rows.push_back(-1);
                        } else {
                            target_rows.push_back(row_target);
                        }
                     } else {
                         auto col_target = std::stoi(lineInput);
                         if ((col_target > col_count) || col_target < 0){
                            target_col.push_back(-1);
                        } else {
                            target_col.push_back(col_target);
                        }
                     }
                 }
             }
         }
    }

    //Making matrix with data list 
    if ((row_count - col_count == 1) || (col_count - row_count == 1)){
        for(int j = 0; j < row_count; j++){
            std::vector<float> rowList = {};
            for (int k = 0; k < col_count; k++){
                rowList.push_back(data_list[(row_count + 1) * j + k]);
            }
            matrix.push_back(rowList);
        }
    } else {
        for(int j = 0; j < row_count; j++){
            std::vector<float> rowList = {};
            for (int k = 0; k < col_count; k++){
                rowList.push_back(data_list[row_count * j + k]);
             }
             matrix.push_back(rowList);
        }
    }

    //Printing Matrix by row and coloumn
    std::cout << std::string("Matrix:" ) << std::endl; 
    for (int row = 0; row < row_count; row ++){
        for (int col = 0 ; col < col_count; col++ ){
            std::cout << std::to_string(matrix[row][col]) << std::string(" "); 
        }
        std::cout << std::string("") << std::endl; 
    }

    std::cout << std::string("\nTargets:" ) << std::endl; 
    for (int target = 0 ; target < target_rows.size(); target++){
        int row = target_rows[target];
        int col = target_col[target];

        if ((target_rows[target] == -1) || (target_col[target] == -1)){
             std::cout << std::string("Target") << std::to_string(target + 1) << std::string(": ERROR TARGET NOT IN MATRIX") << std::endl;
        } else {
            auto target_output =  std::string("Target ") + std::to_string(target + 1) + std::string(": Matrix: [") + std::to_string(row) + std::string("][") + std::to_string(col) + std::string("] =") + std::to_string(matrix[row - 1][col -  1]);
             std::cout << target_output << std::endl;
        }
    }

}